Add support for optional CSS to apply to the notification

NeilRashbrook opened the pull request:
This lets extensions customise the appearance of the notification somewhat (e.g. foreground and background colours).

benbucksch commented on notificationbox/implementation.js:
```diff
@@ -59,7 +59,10 @@ class Notification {
-    this.getNotificationBox().appendNotification(options.label, notificationId, imageURL, options.priority, buttons, callback);
+    let element = this.getNotificationBox().appendNotification(options.label, notificationId, imageURL, options.priority, buttons, callback);
+    if (options.css) {
+      element.setAttribute("style", element.getAttribute("style") + options.css);
```
Even though this works, I think it would be safer to add a semicolor delimiter and space before and after `options.css`. You'd probably end up with double semicolors, but better than none, I'm thinking.

benbucksch commented on notificationbox/implementation.js:
```diff
@@ -57,7 +59,10 @@ class Notification {
-    this.getNotificationBox().appendNotification(this.options.label, this.notificationId, this.imageURL, this.options.priority, buttons, callback);
+    let element = this.getNotificationBox().appendNotification(options.label, notificationId, imageURL, options.priority, buttons, callback);
+    if (options.css) {
+      element.setAttribute("style", element.getAttribute("style") + options.css);
```
```suggestion
      element.setAttribute("style", element.getAttribute("style") + "; " + options.css + "; ");
```

Just a suggestion, not mandatory.

jobisoft replied:
Could we just have a color pair ({ backgroundColor, textColor} ) as input and just set 
```
element.style["background-color"] = backgroundColor;
element.style["color"] = textColor;
```

Would this not be more WebExtension API - ish?

NeilRashbrook replied:
OK I've rewritten the code to that effect and rebased it due to the merge conflict.

NeilRashbrook committed:
Add support for optional CSS to apply to the notification

This lets extensions customise the appearance of the notification somewhat (e.g. foreground and background colours).

benbucksch commented on notificationbox/implementation.js:
```diff
@@ -59,7 +59,10 @@ class Notification {
 
-    this.getNotificationBox().appendNotification(options.label, notificationId, imageURL, options.priority, buttons, callback);
+    let element = this.getNotificationBox().appendNotification(options.label, notificationId, imageURL, options.priority, buttons, callback);
+    for (let key in options.style) {
```
John and I discussed to make a whitelist here. John said he already has the code in mind and wants to add it himself. Thanks, John!

benbucksch replied:
Allowed should be things like `background*`, `color`, `margin*`, `padding*`, `font*` etc.
